aip_urdf_compiler/scripts/compile_xacro.py

```diff
@@ -0,0 +1,484 @@
+#!/usr/bin/python3
+"""
+XACRO Compiler Script.
+
+This script compiles XACRO files for robot sensor configurations. It processes calibration data
+from YAML files and generates XACRO files that define the sensor transforms and configurations
+for a robot's URDF description.
+
+The script handles various types of sensors including cameras, IMUs, LiDARs (Velodyne, Pandar, Livox),
+and radar units, generating appropriate XACRO macros for each sensor type.
+"""
+
+import enum
+import functools
+import os
+from typing import Callable
+from typing import Dict
+from typing import Union
+
+from jinja2 import Template
+import yaml
+
+
+def load_yaml(file_path: str) -> Dict:
+    """
+    Load and parse a YAML file.
+
+    Args:
+        file_path (str): Path to the YAML file
+
+    Returns:
+        Dict: Parsed YAML content or None if parsing fails
+    """
+    with open(file_path, "r") as stream:
+        try:
+            return yaml.safe_load(stream)
+        except yaml.YAMLError as exc:
+            print(exc)
+            return None
```

Comment:
Nits: Returning None makes caller side difficult to handle the error. I prefer to raise the error in load_yaml. For example: 

```suggestion
    try:
        with open(file_path, "r") as stream:
            content = yaml.safe_load(stream)
            if content is None:
                raise ValueError(f"YAML file is empty or invalid: {file_path}")
            if not isinstance(content, dict):
                raise ValueError(f"YAML file must contain a dictionary: {file_path}")
            return content

        except FileNotFoundError:
            raise FileNotFoundError(f"YAML file not found: {file_path}")
        except yaml.YAMLError as exc:
            raise yaml.YAMLError(f"Failed to parse YAML file {file_path}: {str(exc)}")
```

Reply:
I have fixed this in new commit.